Remove THGeneral

peterbell10 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #69419
* __->__ #69041
* #69040
* #69039
* #69038

`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

Differential Revision: [D32872477](https://our.internmc.facebook.com/intern/diff/D32872477)

peterbell10 committed:
Remove THGeneral

`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THGeneral"


`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THGeneral"


`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THGeneral"


`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

[ghstack-poisoned]

peterbell10 committed:
Fix HIP build on "Remove THGeneral"


`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THGeneral"


`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THGeneral"


`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THGeneral"


`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THGeneral"


`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THGeneral"


`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THGeneral"


`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

[ghstack-poisoned]

ngimel commented on aten/src/TH/THConcat.h:
```diff
@@ -0,0 +1,19 @@
+#pragma once
```
should this go to torch/csrc, it's used only there?

peterbell10 committed:
Update on "Remove THGeneral"


`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THGeneral"


`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

[ghstack-poisoned]

peterbell10 committed:
Update on "Remove THGeneral"


`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

[ghstack-poisoned]

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32872477).

ngimel commented:
@peterbell10 can you please also create a macro similar to THArgCheck? It's still used a lot internally, and cannot be easily replaced by TORCH_CHECK, because it accepts `(cond, argN, msg...)`

peterbell10 commented:
@ngimel are you asking for `THArgCheck` to be left as it is, or for something like:
```cpp
#define TORCH_CHECK_ARG(cond, argN, ...) TORCH_CHECK(cond,  "invalid argument ", argN, ": ", __VA_ARGS__)
```

ngimel commented:
`TORCH_CHECK_ARG` is better, it's in line with the rest of the check macros

peterbell10 committed:
Update on "Remove THGeneral"


`TH_CONCAT_{N}` is still being used by THP so I've moved that into
it's own header but all the compiled code is gone.

Differential Revision: [D32872477](https://our.internmc.facebook.com/intern/diff/D32872477)

[ghstack-poisoned]

peterbell10 commented:
@ngimel I've added `TORCH_CHECK_ARG`.

ngimel commented:
Thanks! I'm trying to land previous diffs in the stack (they required some internal changes)

ngimel commented:
@ngimel has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32872477).
